### PR TITLE
For #300: Redis service

### PR DESCRIPTION
**Pull request by paulodamaso**
For #300:
- Added redis service file
- Added redis service configuration on redis installation

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#350](https://github.com/timelesslounge/timelessis/issues/350) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#350](https://github.com/timelesslounge/timelessis/issues/350) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/de64e64e-ac51-4bf9-9aff-3c3f1f8d1919); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: scripts/install/installRedis.sh [hunk 1]
```diff
@@ -4,4 +4,4 @@ wget http://download.redis.io/redis-stable.tar.gz
 tar xvzf redis-stable.tar.gz
 cd redis-stable
 sudo make install
-src/redis-server > /dev/null &
+src/redis-server > /dev/null &
```
Comment (amihaiemil) [inline scripts/install/installRedis.sh]: @paulodamaso Is this line redirecting the output to ``/dev/null``? If yes, why? I guess we need to see any possible logs somehow.

File: scripts/install/deploy/install_redis.sh [hunk 2]
```diff
@@ -30,3 +26,6 @@ if [ "$?" -gt "0" ]; then
 else
   echo "Redis already running"
 fi
+
+sudo cp redis.service /lib/systemd/system/
+sudo systemctl enable redis
```
Comment (amihaiemil) [inline scripts/install/deploy/install_redis.sh]: @paulodamaso Shouldn't these Redis scripts also be added to the Docker image? As I recall, we have setup Docker at some point.

File: scripts/install/deploy/redis.service [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+[Service]
+Type=forking
+ExecStart=redis-server
+SuccessExitStatus=143
+TimeoutStopSec=10
+Restart=on-failure
+RestartSec=5
+
+[Install]
+WantedBy=multi-user.target
```
Comment (amihaiemil) [inline scripts/install/deploy/redis.service]: @paulodamaso new lines at end of files pls

File: scripts/install/deploy/install_redis.sh (hunk 2 above)
Comment (paulodamaso) [reply scripts/install/deploy/install_redis.sh]: @amihaiemil I thought that initially, but @emilianodellacasa said no: https://github.com/timelesslounge/timelessis/issues/300#issuecomment-465233283

**Comment by paulodamaso** [comment]
@amihaiemil Thanks for the comments, please take a look

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@marceloamadeu Take a look at rultor, it is failing for some reason

**Comment by amihaiemil** [comment]
@emilianodellacasa you addressed the comment to the wrong person :)

**Comment by emilianodellacasa** [comment]
@paulodamaso @amihaiemil duh! Sorry, check rultor :)

**Comment by paulodamaso** [comment]
@emilianodellacasa Corrected permission to command, please take a look

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@paulodamaso please check rultor, still not working

**Comment by paulodamaso** [comment]
@emilianodellacasa Looks lile `/var/log` issue; where should I save redis logs?

**Comment by emilianodellacasa** [comment]
@paulodamaso It would be better to have a subdirectory /var/log/redis/ and a redis.log inside. Maybe trying first creating the directory and file, then chaning ownership and permissions, then starting the service?

File: scripts/install/installRedis.sh [hunk 4]
```diff
@@ -4,4 +4,8 @@ wget http://download.redis.io/redis-stable.tar.gz
 tar xvzf redis-stable.tar.gz
 cd redis-stable
 sudo make install
+sudo /var/log/redis
```
Comment (emilianodellacasa) [inline scripts/install/installRedis.sh]: @paulodamaso missing mkdir here

**Comment by paulodamaso** [comment]
@emilianodellacasa Done, I think that it solves the problems

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@paulodamaso looks like redis user and group don't exist

**Comment by paulodamaso** [comment]
@emilianodellacasa It's very strange, I have `redis` user on my machine. Which user is executing this script? To which user should I give permission on `/var/log/redis`?

**Comment by emilianodellacasa** [comment]
@paulodamaso I found the problem, we made some confusion sorry for that! You should'nt work on install/installRedis.sh, because that script install redis on rultor;

Instead, you should work ONLY on install/deploy/install_redis.sh, that is the script that is run on the staging server.

Sorry for noticing it before.

**Comment by paulodamaso** [comment]
@emilianodellacasa Great, so I've reverted he changes, please take a look

File: scripts/install/installRedis.sh [hunk 5]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-sudo apt -y install make gcc libc6-dev tcl
```
Comment (emilianodellacasa) [inline scripts/install/installRedis.sh]: @paulodamaso Don't remove the `-y` please

**Comment by emilianodellacasa** [comment]
@paulodamaso left a comment

**Comment by paulodamaso** [comment]
@emilianodellacasa oops, sorry

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 14 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#350](https://github.com/timelesslounge/timelessis/issues/350) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
The job [#350](https://github.com/timelesslounge/timelessis/issues/350) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by amihaiemil** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/350#issuecomment-466117433))

@amihaiemil This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#350](https://github.com/timelesslounge/timelessis/issues/350) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 42min
  * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will receive 20 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23350%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@amihaiemil thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**